File: Programmers/L2_문자열압축.py
def solution(s):
  answer = len(s)
  for cut in range(1, len(s) // 2 + 1):
    result = 0
    pattern = ''
    cnt = 0
    for i in range(0, len(s), cut):
      newPattern = s[i:i+cut]
      if newPattern == pattern:
        cnt += 1
      else:
        result += len(newPattern)
        if cnt > 1:
          result += len(str(cnt))
        pattern = newPattern
        cnt = 1
    
    if cnt > 1:
      result += len(str(cnt))

    answer = min(answer, result)
  return answer

File: Programmers/L2_문자열압축.js
function solution(s) {
  var answer = s.length;
  for (let cut = 1; cut < parseInt(s.length/2) + 1; cut++) {
    let result = 0
    let pattern = '';
    let cnt = 0;
    for (let i = 0; i < s.length; i += cut) {
      let newPattern = s.slice(i,i+cut);
      if (newPattern == pattern) {
        cnt += 1
      } else {
        result += newPattern.length
        if (cnt > 1) {
          result += String(cnt).length
        }
        pattern = newPattern
        cnt = 1
      }
    }
    if (cnt > 1) {
      result += String(cnt).length
    }
    answer = Math.min(answer, result)
  }
  return answer;
}